app/src/test/java/org/simple/clinic/activity/CloseActivitiesWhenUserIsUnauthorizedTest.kt

```diff
@@ -0,0 +1,234 @@
+package org.simple.clinic.activity
+
+import android.app.Activity
+import com.nhaarman.mockito_kotlin.mock
+import com.nhaarman.mockito_kotlin.never
+import com.nhaarman.mockito_kotlin.reset
+import com.nhaarman.mockito_kotlin.verify
+import com.nhaarman.mockito_kotlin.whenever
+import io.reactivex.subjects.PublishSubject
+import org.junit.Before
+import org.junit.Test
+import org.simple.clinic.user.UserSession
+import org.simple.clinic.util.scheduler.TrampolineSchedulersProvider
+
+class CloseActivitiesWhenUserIsUnauthorizedTest {
+
+  private val userUnauthorizedSubject = PublishSubject.create<Boolean>()
+  private val userSession = mock<UserSession>()
+
+  private val closeActivitiesWhenUserIsUnauthorized = CloseActivitiesWhenUserIsUnauthorized(userSession, TrampolineSchedulersProvider())
+
+  @Before
+  fun setUp() {
+    whenever(userSession.isUserUnauthorized())
+        .thenReturn(userUnauthorizedSubject)
+
+    closeActivitiesWhenUserIsUnauthorized.listen()
+  }
+
+  @Test
+  fun `any created activity that is not TheActivity must be finished if the user gets unauthorized`() {
+    // given
+    val aCreatedTheActivity = mock<TheActivity>()
+    val aCreatedActivity = mock<Activity>()
```

Comment:
Can we name this is such a way to reveal that `TheActivity` is the activity that will never be killed because it hosts the entire application. Whereas `Activity` can be finished because it's just another `Activity`?

Reply:
Maybe that's not required, because the function name captures the intention. Feel free to skip this suggestion.